File: app/src/main/java/com/kode/utair/presentation/ui/adapters/CitiesForecastPagerAdapter.java
package com.kode.utair.presentation.ui.adapters;

import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentPagerAdapter;

import com.kode.utair.presentation.ui.fragments.CityWeatherForecastFragment;

public class CitiesForecastPagerAdapter extends FragmentPagerAdapter {

    public static final int TABS_COUNT = 2;
    public static final int DEPART_CITY_POSITION = 0;
    public static final int ARRIVE_CITY_POSITION = 1;

    private String departCityName;
    private String arriveCityName;

    public CitiesForecastPagerAdapter(FragmentManager fm, String departCityName, String arriveCityName) {
        super(fm);
        this.departCityName = departCityName;
        this.arriveCityName = arriveCityName;
    }

    @Override
    public Fragment getItem(int position) {
        switch (position) {
            case DEPART_CITY_POSITION:
                return CityWeatherForecastFragment.newInstance(departCityName);
            case ARRIVE_CITY_POSITION:
                return CityWeatherForecastFragment.newInstance(arriveCityName);
            default:
                return null;
        }
    }

    @Override
    public int getCount() {
        return TABS_COUNT;
    }


}

File: app/src/main/java/com/utair/presentation/ui/adapters/CitiesForecastPagerAdapter.kt
package com.utair.presentation.ui.adapters

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter
import com.utair.presentation.ui.fragments.CityWeatherForecastFragment

class CitiesForecastPagerAdapter(
        fragmentManager: FragmentManager,
        private val departCityName: String,
        private val arriveCityName: String
) : FragmentPagerAdapter(fragmentManager) {

    override fun getItem(position: Int): Fragment {
        return when (position) {
            DEPART_CITY_POSITION -> CityWeatherForecastFragment.newInstance(departCityName)
            ARRIVE_CITY_POSITION -> CityWeatherForecastFragment.newInstance(arriveCityName)
            else -> throw IllegalStateException()
        }
    }

    override fun getCount(): Int {
        return TABS_COUNT
    }

    companion object {
        const val TABS_COUNT = 2
        const val DEPART_CITY_POSITION = 0
        const val ARRIVE_CITY_POSITION = 1
    }

}